local/include/core/proxy_protocol_util.h

```diff
@@ -0,0 +1,118 @@
+/** @file
+ * Common data structures and definitions for Proxy Verifier tools.
+ *
+ * Copyright 2023, Verizon Media
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#include "swoc/Errata.h"
+#include "swoc/BufferWriter.h"
+#include "swoc/TextView.h"
+#include "swoc/swoc_ip.h"
+
+// PROXY header version
+enum class ProxyProtocolVersion { NONE = 0, V1 = 1, V2 = 2 };
+
+// PROXY header v1 end of header.
+static constexpr swoc::TextView PROXY_V1_EOH{"\r\n"};
+
+// The maximum size of a PROXY header(v1 and v2) without the TLV support.  This
+// is used to specify the data size to peek from the socket.
+static constexpr size_t MAX_PP_HDR_SIZE = 108;
+
+// V1 and V2 header signatures
+using namespace std::literals;
+static const swoc::TextView V1SIG("PROXY");
```

Comment:
hmmm...this isn't properly a signature for proxy protocol. Maybe rename to EXPECTED_PP_PREFIX for now, and add a comment explaining that we don't currently support the other value (`LOCAL` or whatever it is?).

Reply:
Ah, thanks. Yes, I was referring to the command.